packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,302 @@
+| isOpen      | boolean                       | `false`   | Set to `true` when the dialog is visible   |
+| isDraggable | boolean                       | `false`   | Set to `true` to make the dialog draggable |
+| overlay     | _slot_                        | undefined | Dimmed background of dialog                |
+
+- `type` property (dialog variations):
+
+  - `modal`: When this type of dialog is open, the rest of the page is dimmed out and cannot be interacted with. The tab sequence is kept within the dialog and moving the focus outside the dialog will imply closing it. This is the default type of the component.
+
+  - `non-modal`: When a non-modal dialog is open, the rest of the page is not dimmed out and users can interact with the rest of the page. This also implies that the tab focus can move outside the dialog when it reaches the last focusable element.
+
+  - `alert`: are a special type of modal dialogs that interrupts the user's workflow to communicate an important message or ask for a decision. These dialogs are not dismissable, neither by escape key or by clicking outside the dialog.
+
+### DOM
+
+```html
+<div role="dialog" class="fui-dialog">{children}</div>
+```
+
+## Dialog Header
+
+The Header component will expect to have a dialog title/header and will show by default the close (X icon) button.
+
+### DOM
+
+```html
+<div class="fui-dialog-header">
+  <span>{title}</span>
+  <button aria-label="close" />
+</div>
+```
+
+### API
+
+| Property          | Values  | Default     | Purpose                         |
+| ----------------- | ------- | ----------- | ------------------------------- |
+| `closeButton`     | _slot_  | `undefined` | Button that closes the dialog   |
+| `hideCloseButton` | boolean | `false`     | Shows or hides the close button |
 
-_Provide some representative example code that uses the proposed API for the component_
+## Dialog Body
 
-## Variants
+The body is a container where the content of the dialog is rendered. Apart from padding, this component does not have other behaviour.
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+```html
+<div class="fui-dialog-body">{children}</div>
+```
 
-## API
+## Dialog Footer
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+The footer is a container for the actions of the dialog, which must be not more than 3 (primary, secondary and tertiary actions).
 
-## Structure
+### DOM
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+```html
+<div class="fui-dialog-footer">
+  <button>{tertiaryAction}</button>
+  <button>{primaryAction}</button>
+  <button>{secondaryAction}</button>
+</div>
+```
+
+> ⚠️ _Pending issue: should there be any handling for the order of buttons?_
```

Comment:
what about focus ? which button should be focused first when entering action area ?

Reply:
When there are two buttons (primary and secondary) by default the focus will go to the first button (primary). However, the case where there is a tertiary button is still being discussed, because the tertiary button being position all the way to the left also falls as a the first focusable element which is not what we might want. 
I will update this and give more context to this pending issue.